dask/dataframe/io/parquet/arrow.py

```diff
@@ -138,18 +225,55 @@ def read_metadata(
         parts, dataset = _determine_dataset_parts(
             fs, paths, gather_statistics, filters, kwargs.get("dataset", {})
         )
+        # Check if the column-chunk file_path's are set in "_metadata".
+        # If available, we can use the path to sort the row-groups
+        col_chunk_paths = False
+        if dataset.metadata:
+            col_chunk_paths = all(
```

Comment:
Do you have examples where this isn't the case? I think the parquet format implies it must be there (since, the data is not contained in the metadata file)

Reply:
Until this PR, `ArrowEngine` has been (erroneously, of course) producing `"_metadata"` files with missing/`None` file_path metadata.  So, this check is more about handling metadata files that are improperly written (regardless of the standard).

Reply:
got it